website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+        self.wrapperSelector = options.wrapperSelector;  // For encapsulating each implementation of this component in multiple use
+        self.currentLink = ''; // Save the link to compare if a new link is being requested and avoid multiple calls
+        self.reload = m.prop(false); // Gets set to true when treebeard link changes and it needs to be redrawn
+        self.nonLoadTemplate = m.prop(m('.fb-non-load-template.m-md.p-md.osf-box', 'Loading...')); // Template for when data is not available or error happens
+
+        // VIEW STATES
+        self.showInfo = m.prop(true); // Show the info panel
+        self.showSidebar = m.prop(true); // Show the links with collections etc. used in narrow views
+        self.refreshView = m.prop(true); // Internal loading indicator
+        self.allProjectsLoaded = m.prop(false);
+        self.allProjects = m.prop([]);
+        self.loadingNodePages = false; // Since API returns pages of items, this state shows whether filebrowser is still loading the next pages.
+        self.loadingAllNodes = false; // True if we are loading all nodes
+
+        // Load 'All my Projects' and 'All my Registrations'
+        self.systemCollections = [
+            new LinkObject('collection', { path : 'users/me/nodes/', query : { 'related_counts' : true, 'page[size]'  : 12, 'embed' : 'contributors' }, systemCollection : 'nodes'}, 'All My Projects'),
+            new LinkObject('collection', { path : 'users/me/registrations/', query : { 'related_counts' : true, 'page[size]'  : 12, 'embed' : 'contributors'}, systemCollection : 'registrations'}, 'All My Registrations')
+        ];
+        // Initial Breadcrumb for All my projects
+        self.breadcrumbs = m.prop([
+            new LinkObject('collection', { path : 'users/me/nodes/', query : { 'related_counts' : true, 'embed' : 'contributors' }, systemCollection : 'nodes'}, 'All My Projects')
+        ]);
+        // Calculate name filters
+        self.nameFilters = [];
+        // Calculate tag filters
+        self.tagFilters = [];
+
+        // Placeholder for node data
+        self.data = m.prop([]);
+
+        // Activity Logs
+        self.activityLogs = m.prop();
+        self.showMoreActivityLogs = m.prop(null);
+        self.getLogs = function _getLogs (nodeId, link, addToExistingList) {
+            var url = link || $osf.apiV2Url('nodes/' + nodeId + '/logs/', { query : { 'page[size]' : 6, 'embed' : ['nodes', 'user', 'linked_node', 'template_node']}});
+            var promise = m.request({method : 'GET', url : url, config : xhrconfig});
+            self.activityLogs([]); // Empty old logs first;
+            self.showMoreActivityLogs(null);
+            promise.then(function _getLogsThen (result){
+                result.data.map(function(log){
+                    log.attributes.formattableDate = new $osf.FormattableDate(log.attributes.date);
+                    if(addToExistingList){
+                        self.activityLogs().push(log);
```

Comment:
As mentioned in a previous comment, mithril props are supposed to be assigned by calling them with a value. So, rather than treating `map` like `forEach` here, better to create a new array of formatted logs, then assign `self.activityLogs`.

```js
var activityLogs = result.data.map(function(log){
    var formattedLog = $.extend(true, {}, log, {
        attributes: { formattableDate: new $osf.FormattableDate(log.attributes.date); }
    });
    return formattedLog;
});
self.activityLogs(activityLogs);
```